src/expression/definitions/interpolate.ts

```diff
@@ -174,12 +173,13 @@ class Interpolate implements Expression {
         const outputLower = outputs[index].evaluate(ctx);
         const outputUpper = outputs[index + 1].evaluate(ctx);
 
-        if (this.operator === 'interpolate') {
-            return ((interpolate[this.type.kind.toLowerCase()] as any))(outputLower, outputUpper, t); // eslint-disable-line import/namespace
-        } else if (this.operator === 'interpolate-hcl') {
-            return hcl.reverse(hcl.interpolate(hcl.forward(outputLower), hcl.forward(outputUpper), t));
-        } else {
-            return lab.reverse(lab.interpolate(lab.forward(outputLower), lab.forward(outputUpper), t));
+        switch (this.operator) {
+            case 'interpolate':
+                return interpolate[this.type.kind.toLowerCase()](outputLower, outputUpper, t);
```

Comment:
I was planning on commenting here about readability but then saw your deprecation comment, I think using the factory here would make this method a lot more readable.

Reply:
If you find a better way to safeguard this that is not a factory, please do so, this code isn't great at it's current state.

Reply:
Done. 
Please keep in mind that I am doing all this for free in my spare time and don't expect me to fix all the shortcomings in the long existing code. Many of the comments I got from you were about improving the code that was here before me. I believe that I did much more in this PR anyway than the scope of my initial proposal would suggest.